### PR TITLE
Add test cases of query iterator empty expression

### PR DESCRIPTION
**Pull request by NicoYuan1986**
issue: #25561

### CODE REVIEW

File: tests/python_client/testcases/test_query.py [hunk 1]
```diff
@@ -2854,3 +2792,125 @@ def test_count_expression_comparative(self):
...
+        """
+        target: test query iterator with empty expression
+        method: query iterator empty expression with a limit
+        expected: return topK results by order
+        """
+        # 1. initialize with data
+        collection_w, _, _, insert_ids = self.init_collection_general(prefix, True, auto_id=auto_id)[0:4]
+
+        # 2. query with limit
+        collection_w.query_iterator("", limit=limit,
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:2879]: what about query_iterator(limit=limit).

File: tests/python_client/testcases/test_query.py [hunk 2]
```diff
@@ -2854,3 +2792,125 @@ def test_count_expression_comparative(self):
...
+    def test_query_iterator_expr_empty_with_random_pk_pagination(self, limit, offset):
+        """
+        target: test query iterator with empty expression
+        method: create a collection using random pk, query empty expression with a limit
+        expected: return topK results by order
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, with_json=False)[0]
+
+        # 2. generate unordered pk array and insert
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:2897]: Function "init_collection_general " could insert random pk directly setting random_primary_key as True.

File: tests/python_client/testcases/test_query.py [hunk 3]
```diff
@@ -2854,3 +2792,125 @@ def test_count_expression_comparative(self):
...
+
+        # 3. query with empty expr and check the result
+        exp_ids = sorted(unordered_ids)
+        collection_w.query_iterator("", limit=limit, output_fields=[ct.default_string_field_name],
+                                    check_task=CheckTasks.check_query_iterator,
+                                    check_items={"limit": limit, "count": ct.default_nb, "exp_ids": exp_ids})
+
+        # 4. query with pagination
+        exp_ids = sorted(unordered_ids)[offset:]
+        collection_w.query_iterator("", limit=limit, offset=offset, output_fields=[ct.default_string_field_name],
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:2914]: do we have case covering out of max limit or offset? If not, could add a case.

File: tests/python_client/testcases/test_query.py (hunk 2 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_query.py:2897]: I have modified the case using random_primary_key=True.

File: tests/python_client/testcases/test_query.py (hunk 3 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_query.py:2914]: Added.

**Pull request merged**